fedimint-bitcoind/src/lib.rs

```diff
@@ -183,3 +188,50 @@ dyn_newtype_define! {
     #[derive(Clone)]
     pub DynBitcoindRpc(Arc<IBitcoindRpc>)
 }
+
+impl DynBitcoindRpc {
+    pub fn spawn_block_count_update_task(
+        self,
+        task_group: &TaskGroup,
+    ) -> anyhow::Result<watch::Receiver<Option<u64>>> {
+        let (block_count_tx, block_count_rx) = watch::channel(None);
+
+        let mut desired_interval = tokio::time::interval(if is_running_in_test_env() {
+            // In devimint, the setup is blocked by detecting block height changes,
+            // and polling more often is not an issue.
+            debug!(target: LOG_BITCOIND, "Running in devimint, using fast node polling");
+            Duration::from_millis(100)
+        } else {
+            Duration::from_secs(30)
+        });
+
+        task_group.spawn_cancellable("LNv2 module: background update", async move {
```

Comment:
Documentation/task description: Should not be limited to LNv2 if it lives in fedimint-bitcoin. 

Don't we also use something similar in the wallet module? Actually, we could have a `Mutex<Option<watch::Receiver<_>>>` field in `DynBitcoindRpc` that returns an existing handle and spawns a new one if none exists.

That way we could even share the polling between LNv2, LNv1 and the wallet.

Reply:
This PR does move the code from wallet module to bitcoind crate, as a sort of "common".

Reply:
Will change the task description and add some documentation.

As far as sharing the polling, that is possible to do but it would require each module to share the same `DynBitcoinRpc` struct. Currently each module creates their own based on the local config. I'm not sure if we want the client (as opposed to the modules) constructing the `DynBitcoinRpc` since it ties the Client a bit closer to Bitcoin. If we want to do that refactor I think it should be done in a follow up.

Reply:
If we see that over and over many modules do the same thing, then we can think about merging it. Otherwise on an active federation there are just two modules that care.